routing/payment_lifecycle.go

```diff
@@ -180,6 +180,9 @@ func (p *paymentLifecycle) resumePayment(ctx context.Context) ([32]byte,
 		return [32]byte{}, nil, err
 	}
 
+	// Get the payment state.
```

Comment:
can you add a more detailed comment here why this is needed fetch here, maybe something similar as in the commit msg.

Reply:
done!